### PR TITLE
restore webpack defaults for managed/immutablePaths

### PR DESCRIPTION
**Pull request by sokra**
Due to bundling of webpack the default values can't be correctly detected
This restores the webpack defaults

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| buildDuration | 12.2s | 11.5s | -663ms |
| buildDurationCached | 3.8s | 3.5s | -255ms |
| nodeModulesSize | 46.7 MB | 46.7 MB | ⚠️ +3.71 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.087 | 2.079 | -0.01 |
| / avg req/sec | 1197.86 | 1202.55 | +4.69 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.112 | 1.102 | -0.01 |
| /error-in-render avg req/sec | 2249.06 | 2269.05 | +19.99 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| framework-HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 19.4 kB | 19.4 kB | ✓ |
| webpack-HASH.js gzip | 994 B | 994 B | ✓ |
| Overall change | 59.7 kB | 59.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.02 kB | 1.02 kB | ✓ |
| _error-HASH.js gzip | 3.06 kB | 3.06 kB | ✓ |
| amp-HASH.js gzip | 526 B | 526 B | ✓ |
| css-HASH.js gzip | 334 B | 334 B | ✓ |
| hooks-HASH.js gzip | 890 B | 890 B | ✓ |
| index-HASH.js gzip | 262 B | 262 B | ✓ |
| link-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..HASH.js gzip | 331 B | 331 B | ✓ |
| withRouter-HASH.js gzip | 329 B | 329 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 8.54 kB | 8.54 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _buildManifest.js gzip | 390 B | 390 B | ✓ |
| Overall change | 390 B | 390 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| index.html gzip | 561 B | 561 B | ✓ |
| link.html gzip | 569 B | 569 B | ✓ |
| withRouter.html gzip | 557 B | 557 B | ✓ |
| Overall change | 1.69 kB | 1.69 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| buildDuration | 14.6s | 13.5s | -1.2s |
| buildDurationCached | 4.8s | 4.4s | -435ms |
| nodeModulesSize | 46.7 MB | 46.7 MB | ⚠️ +3.71 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| framework-HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 19.4 kB | 19.4 kB | ✓ |
| webpack-HASH.js gzip | 994 B | 994 B | ✓ |
| Overall change | 59.7 kB | 59.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.02 kB | 1.02 kB | ✓ |
| _error-HASH.js gzip | 3.06 kB | 3.06 kB | ✓ |
| amp-HASH.js gzip | 526 B | 526 B | ✓ |
| css-HASH.js gzip | 334 B | 334 B | ✓ |
| hooks-HASH.js gzip | 890 B | 890 B | ✓ |
| index-HASH.js gzip | 262 B | 262 B | ✓ |
| link-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..HASH.js gzip | 331 B | 331 B | ✓ |
| withRouter-HASH.js gzip | 329 B | 329 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 8.54 kB | 8.54 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _buildManifest.js gzip | 390 B | 390 B | ✓ |
| Overall change | 390 B | 390 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _error.js | 17 kB | 17 kB | ✓ |
| 404.html | 2.42 kB | 2.42 kB | ✓ |
| 500.html | 2.41 kB | 2.41 kB | ✓ |
| amp.amp.html | 10.8 kB | 10.8 kB | ✓ |
| amp.html | 1.61 kB | 1.61 kB | ✓ |
| css.html | 1.79 kB | 1.79 kB | ✓ |
| hooks.html | 1.67 kB | 1.67 kB | ✓ |
| index.js | 17.3 kB | 17.3 kB | ✓ |
| link.js | 17.6 kB | 17.6 kB | ✓ |
| routerDirect.js | 17.5 kB | 17.5 kB | ✓ |
| withRouter.js | 17.6 kB | 17.6 kB | ✓ |
| Overall change | 108 kB | 108 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| buildDuration | 10.8s | 10.8s | -12ms |
| buildDurationCached | 4.5s | 4.5s | -52ms |
| nodeModulesSize | 46.7 MB | 46.7 MB | ⚠️ +3.71 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.099 | 2.125 | ⚠️ +0.03 |
| / avg req/sec | 1191 | 1176.56 | ⚠️ -14.44 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.121 | 1.124 | 0 |
| /error-in-render avg req/sec | 2229.67 | 2224.83 | ⚠️ -4.84 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| 677f882d2ed8..HASH.js gzip | 13.3 kB | 13.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 7.26 kB | 7.26 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 60.3 kB | 60.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-HASH.js gzip | 3.74 kB | 3.74 kB | ✓ |
| amp-HASH.js gzip | 536 B | 536 B | ✓ |
| css-HASH.js gzip | 339 B | 339 B | ✓ |
| hooks-HASH.js gzip | 887 B | 887 B | ✓ |
| index-HASH.js gzip | 227 B | 227 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 303 B | 303 B | ✓ |
| withRouter-HASH.js gzip | 302 B | 302 B | ✓ |
| e025d2764813..52f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 9.38 kB | 9.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _buildManifest.js gzip | 420 B | 420 B | ✓ |
| Overall change | 420 B | 420 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| index.html gzip | 613 B | 613 B | ✓ |
| link.html gzip | 621 B | 621 B | ✓ |
| withRouter.html gzip | 606 B | 606 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
Commit: b39ea88e04671181b2a1090cd93e92f4b9bcf7e8

File: packages/next/build/webpack-config.ts [hunk 1]
```diff
@@ -1247,6 +1247,47 @@ export default async function getBaseWebpackConfig(
...
+    // This restores the webpack defaults
+    // @ts-ignore webpack 5
+    webpackConfig.snapshot = {}
+    if (process.versions.pnp === '3') {
+      const match = /^(.+?)[\\/]cache[\\/]jest-worker-npm-[^\\/]+\.zip[\\/]node_modules[\\/]/.exec(
+        require.resolve('jest-worker')
+      )
+      if (match) {
+        // @ts-ignore webpack 5
+        webpackConfig.snapshot.managedPaths = [
```
Comment (ijjk) [inline packages/next/build/webpack-config.ts:1260]: Should this only be added if webpack 5 is being used or does webpack 4 use this as well?

Comment (sokra) [reply packages/next/build/webpack-config.ts:1260]: Only webpack 5

**Review by ijjk — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| buildDuration | 14.4s | 13.5s | -851ms |
| buildDurationCached | 4.3s | 4s | -344ms |
| nodeModulesSize | 46.7 MB | 46.7 MB | ⚠️ +3.71 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.385 | 2.362 | -0.02 |
| / avg req/sec | 1048.41 | 1058.2 | +9.79 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.246 | 1.213 | -0.03 |
| /error-in-render avg req/sec | 2006.58 | 2060.87 | +54.29 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| framework-HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 19.4 kB | 19.4 kB | ✓ |
| webpack-HASH.js gzip | 994 B | 994 B | ✓ |
| Overall change | 59.7 kB | 59.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.02 kB | 1.02 kB | ✓ |
| _error-HASH.js gzip | 3.06 kB | 3.06 kB | ✓ |
| amp-HASH.js gzip | 526 B | 526 B | ✓ |
| css-HASH.js gzip | 334 B | 334 B | ✓ |
| hooks-HASH.js gzip | 890 B | 890 B | ✓ |
| index-HASH.js gzip | 262 B | 262 B | ✓ |
| link-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..HASH.js gzip | 331 B | 331 B | ✓ |
| withRouter-HASH.js gzip | 329 B | 329 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 8.54 kB | 8.54 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _buildManifest.js gzip | 390 B | 390 B | ✓ |
| Overall change | 390 B | 390 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| index.html gzip | 559 B | 559 B | ✓ |
| link.html gzip | 569 B | 569 B | ✓ |
| withRouter.html gzip | 556 B | 556 B | ✓ |
| Overall change | 1.68 kB | 1.68 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| buildDuration | 16.7s | 15.7s | -1s |
| buildDurationCached | 5.3s | 5s | -294ms |
| nodeModulesSize | 46.7 MB | 46.7 MB | ⚠️ +3.71 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| framework-HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 19.4 kB | 19.4 kB | ✓ |
| webpack-HASH.js gzip | 994 B | 994 B | ✓ |
| Overall change | 59.7 kB | 59.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.02 kB | 1.02 kB | ✓ |
| _error-HASH.js gzip | 3.06 kB | 3.06 kB | ✓ |
| amp-HASH.js gzip | 526 B | 526 B | ✓ |
| css-HASH.js gzip | 334 B | 334 B | ✓ |
| hooks-HASH.js gzip | 890 B | 890 B | ✓ |
| index-HASH.js gzip | 262 B | 262 B | ✓ |
| link-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..HASH.js gzip | 331 B | 331 B | ✓ |
| withRouter-HASH.js gzip | 329 B | 329 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 8.54 kB | 8.54 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _buildManifest.js gzip | 390 B | 390 B | ✓ |
| Overall change | 390 B | 390 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _error.js | 16.9 kB | 16.9 kB | ✓ |
| 404.html | 2.42 kB | 2.42 kB | ✓ |
| 500.html | 2.41 kB | 2.41 kB | ✓ |
| amp.amp.html | 10.8 kB | 10.8 kB | ✓ |
| amp.html | 1.61 kB | 1.61 kB | ✓ |
| css.html | 1.79 kB | 1.79 kB | ✓ |
| hooks.html | 1.67 kB | 1.67 kB | ✓ |
| index.js | 17.2 kB | 17.2 kB | ✓ |
| link.js | 17.4 kB | 17.4 kB | ✓ |
| routerDirect.js | 17.4 kB | 17.4 kB | ✓ |
| withRouter.js | 17.4 kB | 17.4 kB | ✓ |
| Overall change | 107 kB | 107 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| buildDuration | 12.4s | 12.6s | ⚠️ +148ms |
| buildDurationCached | 5.2s | 5.1s | -126ms |
| nodeModulesSize | 46.7 MB | 46.7 MB | ⚠️ +3.71 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.431 | 2.456 | ⚠️ +0.02 |
| / avg req/sec | 1028.57 | 1017.98 | ⚠️ -10.59 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.244 | 1.295 | ⚠️ +0.05 |
| /error-in-render avg req/sec | 2010.17 | 1930.77 | ⚠️ -79.4 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| 677f882d2ed8..HASH.js gzip | 13.3 kB | 13.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 7.25 kB | 7.25 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 60.3 kB | 60.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-HASH.js gzip | 3.74 kB | 3.74 kB | ✓ |
| amp-HASH.js gzip | 536 B | 536 B | ✓ |
| css-HASH.js gzip | 339 B | 339 B | ✓ |
| hooks-HASH.js gzip | 887 B | 887 B | ✓ |
| index-HASH.js gzip | 227 B | 227 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 303 B | 303 B | ✓ |
| withRouter-HASH.js gzip | 302 B | 302 B | ✓ |
| e025d2764813..52f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 9.38 kB | 9.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| _buildManifest.js gzip | 420 B | 420 B | ✓ |
| Overall change | 420 B | 420 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js bugfix/managed-paths | Change |
| - | - | - | - |
| index.html gzip | 614 B | 614 B | ✓ |
| link.html gzip | 621 B | 621 B | ✓ |
| withRouter.html gzip | 606 B | 606 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
Commit: bac9a293028919e84a59e698bd892619c26dccea

**Pull request merged**